src/core/algorithms/fd/eulerfd/cluster.cpp

```diff
@@ -0,0 +1,42 @@
+#include "cluster.h"
+
+namespace algos {
+
+void Cluster::ShuffleData(RandomStrategy rand) {
```

Comment:
No, because implementation of std::shuffle depends on STL, but we want have same permutation of array on any platforms and compilers (it is necessary for consistent hash values in test).

Reply:
> No, because implementation of std::shuffle depends on STL, but we want have same permutation of array on any platforms and compilers (it is necessary for consistent hash values in test).

This is an important piece of information, so I believe it is a good idea to leave it as a comment in the code somewhere near this method.